cdap-metadata-spi/src/main/java/io/cdap/cdap/spi/metadata/ScopedNameOfKind.java

```diff
@@ -66,4 +66,10 @@ public int hashCode() {
   public String toString() {
     return '(' + kind.name().toLowerCase() + ')' + super.toString();
   }
+
+  public static ScopedNameOfKind fromString(String s) {
```

Comment:
what will this method do if it is not a well-formed ScopedNameOfKind? It should throw IllegalArgumentException with a meaningful message, including the string that is being parsed. I don't think this method does that? It would throw ArrayOutOfBound exception or similar.